lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -297,10 +309,12 @@ public void collect(int doc) throws IOException {
   // internal versions. If someone will define a constructor with any other
   // visibility, then anyone will be able to extend the class, which is not what
   // we want.
-  private TopFieldCollector(FieldValueHitQueue<Entry> pq, int numHits, int totalHitsThreshold, boolean needsScores) {
+  private TopFieldCollector(FieldValueHitQueue<Entry> pq, int numHits, int totalHitsThreshold,
```

Comment:
We don't need the `totalHitsThreshold` if we have the `hitsThresholdChecker` below